templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -555,3 +555,6 @@ contents:
       # Remove bridges created by ovn-kubernetes
       ovs-vsctl --timeout=30 --if-exists del-br br-int -- --if-exists del-br br-local
     fi
+
+    # Ensure network is online again before network-online.target is reached
+    systemctl try-restart NetworkManager-wait-online.service
```

Comment:
in reload_nm it has ```nm-online -s -t 60; then``` which is the equivalent of NetworkManager-wait-online.service. I'm wondering if the root cause is because this wait does not wait on more than one ip address family unless "may-fail" is added:

```
By default, connections have the ipv4.may-fail and ipv6.may-fail properties set to yes; this means that NetworkManager waits for one of the two address families to complete configuration before considering the connection activated. If you need a specific address family configured before network-online.target is reached, set the corresponding may-fail property to no.
```

So unless we add this may-fail set to no, as long as the connection has an ipv4 addr it will pass the wait-online test and proceed. Thoughts?

Reply:
If that's the case then https://github.com/openshift/cluster-baremetal-operator/pull/239 (+ an equivalent change to the installer) should solve it, by passing `ip=dhcp,dhcp6`. It doesn't by itself though.

I'm assuming that because we only do `activate_nm_conn` _after_ calling `nm-online`, we may not be waiting for those interfaces that weren't active at that time. Then we bring them up and immediately declare success without waiting for DHCP. That's what this is attempting to address.

To date we accidentally passed `ip=dhcp`, but previously the 5s sleep was doing the trick in CI at least. It's certainly possible that we'd need _both_ patches for it to work.

Reply:
Wouldn't nmcli conn up (part of activate_nm_conn) take into account if may-fail=no is set on ipv4/ipv6 and not return until they are both up?

Reply:
I just tested it locally and ^ is correct. If ipv6.may-fail=no and it cant get an IP it times out and fails to activate the connection. I think we should try setting those attributes if we detect the original interface had ipv4 and ipv6 addresses. wdyt?

Reply:
NetworkManage-wait-online is repeatedly failing for me, so adding this is making ovs-configuration fail for me locally.

The default behavior for ip=dhcp,dhcp6 is to create a connection that looks like this:

> [ipv4]
dhcp-timeout=90
dns-search=
method=auto
required-timeout=20000

> [ipv6]
addr-gen-mode=eui64
dhcp-timeout=90
dns-search=
method=auto

Applying this configuration to my local node isn't helping either, so I don't think the CBO patch is going to fix this.

I tried setting may-fail, but it's causing other things to fail. I probably need to try again in a clean environment.

Reply:
Here's a log from a test with `ip=dhcp,dhcp6`: https://paste.centos.org/view/86dc24b4
A theory: the interface we're activating does indeed come up:
```
br-ex:55ed8fd1-0525-416a-8619-ae9520472080:ovs-bridge:1642815247:Sat Jan 22 01\:34\:07 2022:yes:0:no:/org/freedesktop/NetworkManager/Settings/2:yes:br-ex:activated:/org/freedesktop/NetworkManager/ActiveConnection/3::/etc/NetworkManager/system-connections/br-ex.nmconnection
```
But the default interface (I suspect the only one that the kernel command line applies to) is still activating:
```
Wired Connection:fe74bdf4-f9f7-484c-b324-d3d221987983:802-3-ethernet:1642815246:Sat Jan 22 01\:34\:06 2022:yes:0:no:/org/freedesktop/NetworkManager/Settings/1:yes:enp1s0:activating:/org/freedesktop/NetworkManager/ActiveConnection/1::/run/NetworkManager/system-connections/default_connection.nmconnection
```
So we need to wait for _all_ interfaces. _after_ bringing up br-ex.

It's a theory.

Reply:
I tried adding `extra_brex_args+="ipv4.may-fail no ipv6.may-fail no "` to configure-ovs, but for some reason that's causing the ovs ports to fail. o.O